### PR TITLE
Added dynamic form list and export

### LINKED ISSUE #103: Dynamic Form List implementation
Dynamic forms need an own list implementation so only the configured fields are added to the list.

### LINKED ISSUE #108: Export for dynamic forms
Dynamic Form list and export should base on the configured form not an static fieldDescriptors

### PR DESCRIPTION
**Pull request by alexander-schranz**
fixes #108 
fixes #103

### CODE REVIEW

File: DependencyInjection/CompilerPass/DynamicListBuilderCompilerPass.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace L91\Sulu\Bundle\FormBundle\DependencyInjection\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+class DynamicListBuilderCompilerPass implements CompilerPassInterface
```
Comment (wachterjohannes) [inline DependencyInjection/CompilerPass/DynamicListBuilderCompilerPass.php:12]: phpdoc

File: Entity/Dynamic.php [hunk 2]
```diff
@@ -326,4 +336,78 @@ public function getChanged()
...
+    /**
+     * Get field.
+     *
+     * @param $key
+     *
+     * @return string|array
+     */
+    public function getField($key)
+    {
+        return $this->__get($key);
```
Comment (wachterjohannes) [inline Entity/Dynamic.php]: i would do it the other war around (the magic getter calls the "not" magic getter) (:

File: ListBuilder/DynamicListBuilder.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+     * @var string
+     */
+    protected $delimiter;
+
+    /**
+     * DynamicListResolver constructor.
+     *
+     * @param string $delimiter
+     */
+    public function __construct(
```
Comment (wachterjohannes) [inline ListBuilder/DynamicListBuilder.php]: should fit in a single line

File: ListBuilder/DynamicListFactory.php [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+     * @var array
+     */
+    protected $builders;
+
+    /**
+     * DynamicListFactory constructor.
+     *
+     * @param string $defaultBuilder
+     */
+    public function __construct(
```
Comment (wachterjohannes) [inline ListBuilder/DynamicListFactory.php]: should fit in a single line

File: ListBuilder/DynamicListFactory.php [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    /**
+     * Get builder.
+     *
+     * @param $alias
+     *
+     * @return DynamicListBuilderInterface
+     *
+     * @throws \Exception
+     */
+    protected function getBuilder($alias)
```
Comment (wachterjohannes) [inline ListBuilder/DynamicListFactory.php]: make parameter optional.

**Comment by alexander-schranz** [comment]
@wachterjohannes PR updated

**Pull request merged**